bwdata.py

```diff
@@ -0,0 +1,805 @@
+            next_mentions = self._get_mentions_page(params, params["page"])
+
+            if len(next_mentions) > 0:
+                all_mentions += next_mentions
+
+                if self.console_report:
+                    print("Page " + str(params["page"]) + " of " + self.resource_type + " " + name + " retrieved")
+            else:
+                break
+
+            params["page"] += 1
+
+        if self.console_report:
+            print(str(len(all_mentions)) + " mentions downloaded")
+        return all_mentions
+
+    def num_mentions(self, name=None, startDate=None, **kwargs):
+        """
+        Retrieves a count of the mentions in a given timeframe.
+
+        Args:
+            name:       You must pass in a query / group name (string).
+            startDate:  You must pass in a start date (string).
+            kwargs:     All other filters are optional and can be found in filters.py.
+
+        Returns:
+            A count of the mentions in a given timeframe.
+        """
+        params = self._fill_params(name, startDate, kwargs)
+        return self.project.get(endpoint="data/mentions/count", params=params)
+
+    def get_chart(self, name=None, startDate=None, y_axis=None, x_axis=None, breakdown_by=None, **kwargs):
+        """
+        Retrieves chart data. 
+
+        Args:
+            name:           You must pass in a query / group name (string).
+            startDate:      You must pass in a start date (string).
+            y_axis:         Pass in the y axis of your chart (string in camel case). See Brandwatch app dropdown menu "For (Y-Axis)" for options
+            x_axis:         Pass in the x axis of your chart (string in camel case). See Brandwatch app dropdown menu "Show (X-Axis)" for options.
+            breakdown_by:   Pass in breakdown_by (string in camel case). See Brandwatch app dropdown menu "Breakdown by" for options.
+            kwargs:         You must pass in name (query name/group) and startDate (string).  Additionally, if you x_axis or breakdown_by consists of categories or tags you must pass in dim1Args or dim2Args, respectively, which should be a list of the names of those cats/tags. All other filters are optional and can be found in filters.py.
+        
+        Returns:
+            A dictionary representation of the specified chart
+
+        Raises:
+            KeyError:       If you fail to pass in x_axis, y_axis or breakdown_by.
+
+        """
+        if not (y_axis and x_axis and breakdown_by):
+            raise KeyError("You must pass in an y_axis, x_axis and breakdown_by")
+
+        params = self._fill_params(name, startDate, kwargs)
+        if "dim1Args" in params:
+            params["dim1Args"] = self._name_to_id(x_axis, params["dim1Args"])
+        if "dim2Args" in params:
+            params["dim2Args"] = self._name_to_id(breakdown_by, params["dim2Args"])
+
+        return self.project.get(endpoint="data/"+y_axis+"/"+x_axis+"/"+breakdown_by, params=params)
```

Comment:
It returns a lot of stuff, but I'm kind of in favor of leaving the whole dictionary because it recaps what you're aggregating and what the x and y axes are. I've pasted an example below for your reference. Thoughts @pwsiegel?

{'aggregate': 'volume',
 'dimension1': 'categories',
 'dimension2': 'days',
 'results': [{'data': {},
   'id': xxxxxxx,
   'name': "Example Chart",
   'values': [{'data': {},
     'id': '2016-08-24T00:00:00.000+0000',
     'name': '2016-08-24 00:00:00.0',
     'value': 5},
    {'data': {},
     'id': '2016-09-03T00:00:00.000+0000',
     'name': '2016-09-03 00:00:00.0',
     'value': 6},
    {'data': {},
     'id': '2016-07-05T00:00:00.000+0000',
     'name': '2016-07-05 00:00:00.0',
     'value': 10},.......